File: src/www/__test__/macros/TitleMacro.js
import AbstractRegExpMacro from "./AbstractRegExpMacro";

export default class TitleMacro extends AbstractRegExpMacro {
  constructor() {
    super(/^###+\s/);
  }

  async interpretMatch(match, { source }, { set }) {
    set("currentTitle", source);
  }
}

File: src/test/java/com/drpicox/game/macros/TitleMacro.java
package com.drpicox.game.macros;

import com.drpicox.game.interpreter.Context;
import com.drpicox.game.interpreter.Instruction;
import org.springframework.stereotype.Component;

import java.util.regex.MatchResult;

@Component
public class TitleMacro extends AbstractRegExpMatcher {

    public TitleMacro() {
        super("###[^#].*");
    }

    @Override
    protected void interpretMatch(MatchResult match, Instruction instruction, Context context) {
        context.set("currentTitle", instruction.getSource());
    }
}
